packages/vscode/src/autocomplete.ts

```diff
@@ -109,6 +109,20 @@ export async function registerAutoComplete(
       if (!code)
         return null
 
+      const offset = window.activeTextEditor!.document.offsetAt(position)
+
+      const afterCode = code.slice(offset)
+        .replace(/"[^"<>{}]*"/g, '')
+        .replace(/\{[^}]*\}/g, '')
+        .replace(/'[^'<>{}]*'/g, '')
+      // get style range
+      const start = code.indexOf('<style')
+      const end = code.lastIndexOf('</style')
```

Comment:
Could we extract this as a function like `shouldProvideAutocomplete` and add some unit test/snapshots to this so we can maintain it easier?

Reply:
sure